### PR TITLE
add status for applicant class

### PR DESCRIPTION
**Pull request by SethCKL**
to be used in updating the applicant's status when a job offer is accepted.

the original applicant constructor includes a default status, which has the value "Available"

should the applicant need to accept a job offer, the status can be changed using setStatus and a new applicant object is returned.

### CODE REVIEW

File: src/main/java/seedu/address/model/applicant/Status.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    /**
+     * Constructs a {@code HiredStatus}.
+     * Initialised as available as every Applicant added to HireLah is applying for a position.
+     */
+    public Status() {
+        value = "Available";
+    }
+
+    public Status(String positionName) {
+        value = positionName;
```
Comment (likeabowx) [inline src/main/java/seedu/address/model/applicant/Status.java:19]: Maybe the value of the status can be something like `"Hired as" + positionName` for better clarify?

**Review by likeabowx — CHANGES_REQUESTED** [review]
Good job implementing the status for applicant as e discussed previously. Just a small thing, I understand that interview has a `Status` class as well which is different from this status for applicant. Maybe it'll be better to name is as `ApplicantStatus` to order confusion? Or do you think it's ok since it's for different data type?

File: src/main/java/seedu/address/model/applicant/Status.java (hunk 1 above)
Comment (SethCKL) [reply src/main/java/seedu/address/model/applicant/Status.java]: Good points. I think `ApplicantStatus` might be still be abit vague so i decided to rename the class to `HiredStatus` as the main purpose of this status is to mark if the applicant has been hired or not as a result of obtaining the job offer.

Also agreed on adding a preamble of "Hired as" to the value for better clarity

**Review by likeabowx — APPROVED** [review]
LGTM

**Pull request merged**